test/unit/util/define-dynamic-block.test.js

```diff
@@ -19,50 +28,73 @@ const penIconURI = 'data:image/svg+xml;base64,fake_pen_icon_svg_base64_data';
 
 const testBlockInfo = {
     commandWithIcon: {
-        blockType: BlockType.COMMAND,
         blockIconURI: penIconURI,
-        text: 'command with icon'
+        info: {
+            blockType: BlockType.COMMAND,
+            text: 'command with icon'
+        }
     },
     commandWithoutIcon: {
-        blockType: BlockType.COMMAND,
-        text: 'command without icon'
+        info: {
+            blockType: BlockType.COMMAND,
+            text: 'command without icon'
+        }
     },
     terminalCommand: {
-        blockType: BlockType.COMMAND,
-        isTerminal: true,
-        text: 'terminal command'
+        info: {
+            blockType: BlockType.COMMAND,
+            isTerminal: true,
+            text: 'terminal command'
+        }
     },
     reporter: {
-        blockType: BlockType.REPORTER,
-        text: 'reporter'
+        info: {
+            blockType: BlockType.REPORTER,
+            text: 'reporter'
+        }
     },
     boolean: {
-        blockType: BlockType.BOOLEAN,
-        text: 'Boolean'
+        info: {
+            blockType: BlockType.BOOLEAN,
+            text: 'Boolean'
+        }
     },
     hat: {
-        blockType: BlockType.HAT,
-        text: 'hat'
+        info: {
+            blockType: BlockType.HAT,
+            text: 'hat'
+        }
+    },
+    blockWithCustomContextMenu: {
+        info: {
+            blockType: BlockType.COMMAND,
+            text: 'block with custom context menu',
+            customContextMenu: [{
+                text: 'Context Menu Item',
+                callback: () => {}
+            }]
+        }
     }
 };
 
 // similar to goog.mixin from the Closure library
-const mixin = function (target, source) {
+const mixin = (target, source) => {
     for (const x in source) {
         target[x] = source[x];
     }
 };
 
 class MockBlock {
-    constructor (blockInfo, extendedOpcode) {
+    constructor (staticBlockInfo, extendedOpcode) {
         // mimic Closure-style inheritance by mixing in `defineDynamicBlock` output as this instance's prototype
         // see also the `Blockly.Block` constructor
-        const prototype = defineDynamicBlock(MockScratchBlocks, categoryInfo, blockInfo, extendedOpcode);
+        const ddb = defineDynamicBlock.bind(this);
+        const prototype = ddb(MockScratchBlocks, categoryInfo, staticBlockInfo, extendedOpcode);
```

Comment:
Seems like these two lines could be replaced with:
```js
const prototype = defineDynamicBlock.call(this, MockScratchBlocks, categoryInfo, staticBlockInfo, extendedOpcode);
```

Reply:
Oh no, that was an accidental remnant of me trying to make the tests work! I'm going to change it back to the one prototype line that it was before.

Reply:
Fixed!